client/packages/common/src/ui/layout/tables/utils/ColumnUtils.ts

```diff
@@ -0,0 +1,110 @@
+import { useTranslation } from '@common/intl';
+import { RecordWithId } from '@common/types';
+import { ArrayUtils } from '@common/utils';
+
+type ColumnProperty<T> = {
+  path: T[];
+  default?: string;
+};
+
+export const useColumnUtils = () => {
+  const t = useTranslation();
+
+  /**
+   * This method tries to extract a single common field value from the input data, using a list of given object paths.
+   *
+   * Data is only extracted from the first matching path, (a path matches if path[0] exists in the input data)
+   *
+   * If the path contains one or more arrays multiple field values are collected.
+   * If all field values are the same this common field value is returned.
+   * If field values differ the default value is returned instead.
+   *
+   * If a path points to a single field the matching field value is returned. If there is no match the default is returned.
+   */
+  const getValue = <T extends object>(
+    row: T,
+    property: ColumnProperty<KeysOfUnion<T>>
+  ): unknown => {
+    if (property.path.length === 0) {
+      return undefined;
+    }
+
+    const path = property.path[0] as keyof T;
+    if (path === undefined || path === null) return undefined;
+    if (!(path in row)) return undefined;
+    const key = property.path[1] as keyof T;
+
+    if (Array.isArray(row[path])) {
+      if (property.path.length < 2) return undefined;
```

Comment:
Nit: currently only path.length < 3 is supported, you could check for this and print a warning?